// Copyright 2020 Google LLC. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package tfx.orchestration;

import "ml_metadata/proto/metadata_store.proto";

message ArtifactList {
  repeated ml_metadata.Artifact artifacts = 1;
}

message DriverOutput {
  // Users use this map to store the proposed change to output artifacts of the
  // upcoming execution.
  map<string, ArtifactList> output_artifacts = 1;
  // TODO(b/166314342): Investigate the necessity to add exec_properties before
  // adding it.
}
